kfs-core/src/main/java/edu/arizona/kfs/gl/batch/service/impl/BudgetAdjustmentCashTransferServiceImpl.java

```diff
@@ -0,0 +1,596 @@
+     */
+    @Override
+    public void extractAndSaveBudgetAdjustmentEntries() {
+    	     
+    	String inputFile = batchFileDirectoryName + File.separator + GeneralLedgerConstants.BatchFileSystem.BUDGET_ADJUSTMENT_OUTPUT_FILE + GeneralLedgerConstants.BatchFileSystem.EXTENSION;
+        String errorFile = batchFileDirectoryName + File.separator + GeneralLedgerConstants.BatchFileSystem.CASH_TRANSFER_TRANSACTIONS_ERROR_OUTPUT_FILE + GeneralLedgerConstants.BatchFileSystem.EXTENSION; 
+        
+        FileReader INPUT_GLE_FILE = null;
+        PrintStream OUTPUT_ERR_FILE_ps;
+        String GLEN_RECORD;
+        BufferedReader INPUT_GLE_FILE_br;
+        try {
+        	INPUT_GLE_FILE = new FileReader(inputFile);
+        	OUTPUT_ERR_FILE_ps = new PrintStream(errorFile);
+        }
+        catch (FileNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+         
+        INPUT_GLE_FILE_br = new BufferedReader(INPUT_GLE_FILE);
+        int line = 0;
+        int reportBudgetAdjustDocLoaded = 0;
+        int reportBudgetAdjustDocErrors = 0;
+        int maxSequenceId = 0;
+        String saveDocNumber = new String("");
+        		
+        try {
+        	while ((GLEN_RECORD = INPUT_GLE_FILE_br.readLine()) != null) {
+        		if (!org.apache.commons.lang.StringUtils.isEmpty(GLEN_RECORD) && !org.apache.commons.lang.StringUtils.isBlank(GLEN_RECORD.trim())) {
+                	 line++;
+                	 GLEN_RECORD = org.apache.commons.lang.StringUtils.rightPad(GLEN_RECORD, 183, ' ');
+                	 OriginEntryFull originEntry = new OriginEntryFull();
+                	 List<Message> parsingError = new ArrayList();
+                	 parsingError = originEntry.setFromTextFileForBatch(GLEN_RECORD, line);
+                	 
+                	 if (parsingError.size() > 0) {
+                		 //parsing error, write error and continue	   
+                		 reportWriterService.writeError(originEntry, parsingError);
+                		 createOutputEntry(GLEN_RECORD, OUTPUT_ERR_FILE_ps);
+                		 reportBudgetAdjustDocErrors++;
+                		 continue;
+                     }
+                     
+                     if (isBudgetAdjustmentTransaction(originEntry)) {
+                    	 
+                    	//check to make sure that Account has Income Stream information
+                    	 Account baAccount = originEntry.getAccount();    	 
+                         if (ObjectUtils.isNotNull(baAccount)) {                        
+                        	 if (ObjectUtils.isNull(baAccount.getIncomeStreamFinancialCoaCode()) || ObjectUtils.isNull(baAccount.getIncomeStreamAccountNumber())) {             	
+	                        	 //no income stream info, write error and continue	                        	
+	                             Message errorMsg = new Message("No Account Income Stream information found for this record.", Message.TYPE_FATAL);
```

Comment:
Lets discuss the pros and cons of creating constants for batch error messages that won't be used anywhere else.